data_preprocessing.py

```diff
@@ -2,33 +2,66 @@
 # Purpose: Code for preparing real data for further analysis
 import nibabel as nib
 import numpy as np
-import matplotlib.pyplot as plt
+
 
 data_dir = '/Users/is321/Documents/Data/qBold/hyperv_data/'
 
-subjects = ['CISC17352', 'CISC17543', 'CISC17987', 'CISC19890']
 
-img = nib.load(data_dir + subjects[3] + '/hyperv_ase.nii.gz')
-data = img.get_fdata()
 
 
-def estimate_noise_level(data):
+def estimate_noise_level():
+    import matplotlib.pyplot as plt
+
+    subjects = ['CISC17352', 'CISC17543', 'CISC17987', 'CISC19890']
+
+    dir_name = data_dir + subjects[2]
+    basename = 'baseline_ase'
+    brain_mask = dir_name + '/mask_' + basename + '_mask.nii.gz'
+    img = nib.load(dir_name + '/'+basename+'.nii.gz')
+    data = img.get_fdata()
+
+    mask = nib.load(brain_mask)
+    mask = mask.get_fdata()
+
+    mask = mask.reshape(-1)
+
+    within_mask_data = data.reshape(-1, 11)[mask>0, :]
+    outside_mask_data = data.reshape(-1, 11)[mask==0, :]
+
+    within_mask_mean = np.mean(within_mask_data, 0)
+    outside_mask_std = np.std(outside_mask_data, 0)
+
+
     # Estimate the SNR from the corner of the image (0 true signal)
-    corner = data[1:18, 1:18, :, :]
+    corner_vals = 12
+    corner = data[1:corner_vals, 1:corner_vals, :, :]
+    corner2 = data[-corner_vals:-1, -corner_vals:-1, :, :]
+    corner3 = data[-corner_vals:-1, 1:corner_vals, :, :]
+    corner4 = data[1:corner_vals, -corner_vals:-1, :, :]
+    corner = np.concatenate([corner, corner2, corner3, corner4], 0)
     # Reshape
     corner = corner.reshape(-1, 11)
+
+    corner_std = np.std(corner[:, :], 0)
+    snr = within_mask_mean / corner_std
+    norm_snr = snr / snr[3]
```

Comment:
Maybe we could then save this for use where aforementioned stdd range is hardcoded